WcaOnRails/app/views/competitions/_index_table.html.erb

```diff
@@ -1,8 +1,22 @@
 <ul class="list-group">
   <li class="list-group-item"><strong><%= title %></strong></li>
   <% competitions.each_with_index do |competition, index| %>
+    <% if index > 0 && competition.year != competitions[index - 1].year %>
+      <li class="list-group-item break"><%= competition.year %></li>
+    <% end %>
     <li class="list-group-item <%= competition.is_over? ? "past" : "not-past" %>">
-      <span class="date"><%= date_range(competition.start_date, competition.end_date, separator: '-', show_year: false) %>, <%= competition.year %></span>
+      <span class="date">
+        <% if competition.in_progress? %>
+        <%=  icon('hourglass-end') %>
+        <% elsif competition.is_over? %>
+        <%=  icon('hourglass-o') %>
+        <% elsif competition.start_date <= 1.month.from_now %>
+        <%=  icon('hourglass-half') %>
+        <% else %>
+        <%=  icon('hourglass-start') %>
+        <% end %>
+        <%= date_range(competition.start_date, competition.end_date, separator: '-', show_year: false) %>, <%= competition.year %>
+      </span>
       <span class="competition-info">
         <p class="competition-link"><%= link_to competition.name, competition_path(competition) %></p>
         <p class="location"><strong><%= competition.country.name %></strong>, <%= competition.cityName %></p>
```

Comment:
Oh, I just realised I'm dumb and didn't write that the flags should be national flags :P
That's why I had the question mark there, because I don't know if we have the little flags saved somewhere.